snapshot/README.md

```diff
@@ -0,0 +1,139 @@
+--- 
+layout: pattern
+title: Snapshot
+folder: snapshot
+permalink: /patterns/snapshot/
+categories:
+- creational 
+  language: 
+  tags:
+- Data access
+- Gang of Four
+---
+
+## Name / classification
+
+Snapshot.
+
+## Intent
+
+Create a view of an object at a point in time.
+
+## Explanation
+
+Real world example
+
+> Many people have phone numbers with details attached. A phone book, where registered phone 
+> numbers with details are relieved are snapshots of the history of phone numbers, and their 
+> associated details.
+
+In plain words
+
+> A snapshot is a copy of an object as it is at a point in time.
+
+Wikipedia says
+
+> The memento pattern is a software design pattern that exposes the private internal state of an 
+> object. One example of how this can be used is to restore an object to its previous state 
```

Comment:
Memento is not the same as the snapshot pattern right? This does not make that very clear.